src/test/groovy/ApmBasePipelineTest.groovy

```diff
@@ -493,6 +493,12 @@ class ApmBasePipelineTest extends DeclarativePipelineTest {
       def script = loadScript('vars/nodeArch.groovy')
       return script.call()
     })
+    helper.registerAllowedMethod('githubCheck', [Map.class], { m ->
+      if(m.name.equalsIgnoreCase('failed')){
+        updateBuildStatus('FAILURE')
```

Comment:
The fact that an exception is thrown doesn't automatically fail the build?

Reply:
Yes, this is just mocking a failure during the UTs, so exception is not handled in the JenkinsPipelineUnit at runtime, so I explicitly set the status in the default mock. If I recall, there is no a helper function to simplify this :(

Reply:
Interesting! Thanks for the explanation.